File: rxpanda/src/main/java/com/pandaq/rxpanda/requests/okhttp/DeleteRequest.java
package com.pandaq.rxpanda.requests.okhttp;

import com.pandaq.rxpanda.RxPanda;
import com.pandaq.rxpanda.observer.ApiObserver;
import com.pandaq.rxpanda.requests.okhttp.base.HttpRequest;
import io.reactivex.Observable;

import java.lang.reflect.Type;

/**
 * Created by huxinyu on 2019/3/16.
 * Email : panda.h@foxmail.com
 * <p>
 * Description :request in delete method
 */
public class DeleteRequest extends HttpRequest<DeleteRequest> {
    public DeleteRequest(String url) {
        super(url);
    }

    @Override
    protected <T> Observable<T> execute(Type type) {
        return mApi.delete(url, localParams)
                .doOnSubscribe(disposable -> {
                    if (tag != null) {
                        RxPanda.manager().addTag(tag, disposable);
                    }
                })
                .compose(httpTransformer(type));
    }

    @SuppressWarnings("unchecked")
    @Override
    protected void execute(ApiObserver callback) {
        if (tag != null) {
            RxPanda.manager().addTag(tag, callback);
        }
        this.execute(getType(callback)).subscribe(callback);
    }
}

File: rxpanda/src/main/java/com/pandaq/rxpanda/requests/okhttp/DeleteRequest.kt
package com.pandaq.rxpanda.requests.okhttp

import com.pandaq.rxpanda.requests.okhttp.base.HttpRequest
import okhttp3.ResponseBody

/**
 * Created by huxinyu on 2019/3/16.
 * Email : panda.h@foxmail.com
 *
 *
 * Description :request in delete method
 */
class DeleteRequest(url: String) : HttpRequest<DeleteRequest>(url) {

    override suspend fun execute(): ResponseBody? {
        return mApi?.delete(url, localParams)
    }

}